src/uct/cuda/cuda_copy/cuda_copy_iface.h

```diff
@@ -49,22 +51,36 @@ typedef struct uct_cuda_copy_queue_desc {
 } uct_cuda_copy_queue_desc_t;
 
 
+typedef struct uct_cuda_copy_ctx_rsc {
+    /* CUDA context handle */
+    CUcontext                  ctx;
+    /* CUDA context id */
+    unsigned long long         ctx_id;
+    /* pool of cuda events to check completion of memcpy operations */
+    ucs_mpool_t                event_mp;
+    /* stream used to issue short operations */
+    CUstream                   short_stream;
+    /* array of queue descriptors for each src/dst memory type combination */
+    uct_cuda_copy_queue_desc_t queue_desc[UCS_MEMORY_TYPE_LAST]
+                                         [UCS_MEMORY_TYPE_LAST];
+} uct_cuda_copy_ctx_rsc_t;
+
+
+/* Hash map for CUDA context resources. The key is the CUDA context Id. */
+KHASH_INIT(cuda_copy_ctx_rscs, unsigned long long, uct_cuda_copy_ctx_rsc_t*, 1,
```

Comment:
can't we keep `uct_cuda_copy_ctx_rsc_t` in the hash (not a pointer) to avoid malloc/free in our code?
afaiu it was like that before in this PR, what is the reason to change?

Reply:
the event queue contains list head which gets corrupted when hash is resized